multimodal/src/autogluon/multimodal/data/infer_types.py

```diff
@@ -326,7 +355,7 @@ def infer_problem_type_output_shape(
     """
     if provided_problem_type is not None:
         if provided_problem_type == MULTICLASS or provided_problem_type == BINARY:
```

Comment:
In the following, we also tested for `provided_problem_type == REGRESSION`. Actually, it will never reach this condition because we tested via `if provided_problem_type == MULTICLASS or provided_problem_type == BINARY:`. Thus, we should change the if condition to 

```python
if provided_problem_type in [MULTICLASS, BINARY, REGRESSION]:
```

Reply:
`if provided_problem_type == REGRESSION` isn't under the condition `if provided_problem_type == MULTICLASS or provided_problem_type == BINARY`. It's under the condition `if provided_problem_type is not None`.